ddsketch/encoding/encoding.go

```diff
@@ -60,6 +63,23 @@ func DecodeUvarint64(b *[]byte) (uint64, error) {
 	}
 }
 
+// Uvarint64Size returns the number of bytes that EncodeUvarint64 encodes a
+// 64-bit unsigned integer into.
+func Uvarint64Size(v uint64) int {
+	return uvarint64Sizes[bits.LeadingZeros64(v)]
```

Comment:
Is this faster than a switch statement with 4 cases? I guess this implementation is branchless and probably uses assembly under the hood, just seems a little complicated but if it works im cool with it

Reply:
I think we'd need 9 cases, each for a range, so I'm not convinced it'd be better. We can also calculate the size directly without any branch, as so:

```
func Uvarint64Size(v uint64) int {
	zeros := bits.LeadingZeros64(v)
	x := (79 - (zeros + zeros>>3)) >> 3
	return x | int(uint(x-1)>>63) // == min(1, x)
}
```
But benchmarks show that it is more costly, so I'll keep the precomputed array.